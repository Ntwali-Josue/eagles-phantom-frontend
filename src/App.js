// class App extends Component {
//   render() {
//     return (
//       <div>
//         <h2>The following is the counter of number.</h2>
//         <h3 id="value" aria-label="value" >value: {this.props.value}</h3>
//         <button id="incrementButton" aria-label="incrementButton" name="increment" onClick={this.props.increment}>Increment</button>
//         <button id="decrementButton" aria-label="decrementButton" name="decrement" onClick={this.props.decrement}>decrement</button>
//       </div>
//     );
//   }
// }

// const mapStateToProps = (state) => {
//   return {
//     value: state.count.value,
//   };
// };

// const mapDispatchToProps = (dispatch) => {
//   return {
//     increment: () => dispatch(actions.increment()),
//     decrement: () => dispatch(actions.decrement()),
//   };
// };

// export default connect(mapStateToProps, mapDispatchToProps)(App);

import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Home from './components/landingPage/Home';
import AboutUs from './components/AboutUs';
import ContactUs from './components/ContactUs';
import Login from './components/Login';
import Error from './components/Error';


class App extends Component{
    render(){
        return (
            <div>
              <Switch>
                  <Route  path='/' component={Home} exact />
                  <Route path='/about' component={AboutUs}/>
                  <Route path='/contact' component={ContactUs} />
                  <Route path='/login' component={Login}/>
                  <Route component={Error}/>
              </Switch>
            </div>
        );
    };
}
export default App;

