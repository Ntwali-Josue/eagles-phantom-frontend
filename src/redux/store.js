import {createStore,compose,combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import counterReducer from "./reducers/counter";
import { composeWithDevTools} from 'redux-devtools-extension';
import promise from 'redux-promise-middleware';
import allReducers from './reducers';

const rootReducer = combineReducers({
  count: counterReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const middleware = applyMiddleware(thunk, promise)
const store =createStore(allReducers, {}, composeWithDevTools(middleware));

export default (store,createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(thunk))
));
