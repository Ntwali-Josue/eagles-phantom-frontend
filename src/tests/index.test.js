import React from 'react';
import { render } from "@testing-library/react";
import App from '../App';
import { Provider } from "react-redux";
import userEvent from '@testing-library/user-event';
import store from "../redux/store.js";


const AppComponent = () => {
  return render(
    <Provider store={store} >
    <App />
  </Provider>
  )
}
describe("APP", () => {
  it("should render the component", () => {
      const { asFragment } = AppComponent();
      expect(asFragment(<App />)).toMatchSnapshot()
    });
 it("increment", () => {
    const { getByLabelText } = AppComponent();
    const incrementButton = getByLabelText("incrementButton")
    const value = getByLabelText("value")
    const before = parseInt(value.innerHTML, 10);
    userEvent.click(incrementButton)
    expect(parseInt(value.innerHTML)).toBe(before + 1)
  })
it("decrement", () => {
    const { getByLabelText } = AppComponent();
    const decrementButton = getByLabelText("decrementButton")
    const value = getByLabelText("value")
    const before = parseInt(value.innerHTML, 10);
    userEvent.click(decrementButton)
    expect(parseInt(value.innerHTML)).toBe(before - 1)
  })
})
