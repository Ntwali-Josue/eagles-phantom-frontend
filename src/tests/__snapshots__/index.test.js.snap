// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should render the component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello!! Welcome to phantom App.
    </h1>
    <h2>
      The following is the counter of number.
    </h2>
    <h3
      aria-label="value"
      id="value"
    >
      value: 0
    </h3>
    <button
      aria-label="incrementButton"
      id="incrementButton"
      name="increment"
    >
      Increment
    </button>
    <button
      aria-label="decrementButton"
      id="decrementButton"
      name="decrement"
    >
      decrement
    </button>
  </div>
</DocumentFragment>
`;
